include/xtensor/xmasked_value.hpp

```diff
@@ -0,0 +1,646 @@
+/***************************************************************************
+* Copyright (c) 2017, Johan Mabille, Sylvain Corlay, Wolf Vollprecht and   *
+* Martin Renou                                                             *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XTENSOR_XMASKED_VALUE_HPP
+#define XTENSOR_XMASKED_VALUE_HPP
+
+#include "xtl/xoptional.hpp"
+
+#include "xutils.hpp"
+
+namespace xt
+{
+    template <class T, class B = bool>
+    class xmasked_value;
+}
+
+// Monkey patch on the xtl::common_optional_t implementation
+namespace xtl
+{
+    namespace detail
+    {
+        template <class T1, class T2, class B2>
+        struct common_optional_impl<T1, xt::xmasked_value<T2, B2>>
+            : common_optional_impl<T1, T2>
+        {
+        };
+
+        template <class T1, class B1, class T2>
+        struct common_optional_impl<xt::xmasked_value<T1, B1>, T2>
+            : common_optional_impl<T1, T2>
+        {
+        };
+    }
+}
+
+namespace xt
+{
+    template <class T>
+    inline xmasked_value<T, bool> masked() noexcept
+    {
+        return xmasked_value<T, bool>(T(0), false);
+    }
+
+    namespace detail
+    {
+        template <class E>
```

Comment:
one could switch to inherit from the new is_crtp_base_of

https://github.com/QuantStack/xtensor/pull/1041/files

Reply:
xmasked_value is not a crtp class :)